source/heating_model/constant_tidal_heating.cc

```diff
@@ -0,0 +1,126 @@
+    MaterialModel::MaterialProperties::Property
+    ConstantTidalHeating<dim>::
+    get_required_properties () const
+    {
+      return MaterialModel::MaterialProperties::none;
+    }
+
+
+
+    template <int dim>
+    void
+    ConstantTidalHeating<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Heating model");
+      {
+        prm.enter_subsection("Constant tidal heating");
+        {
+          prm.declare_entry ("Constant tidal frequency", "2e-5",
+                             Patterns::Double (0),
+                             "The tidal frequency of moon. "
+                             "Reference value is for Europa, ~3.55 days. "
+                             "Units: 1/s.");
+          prm.declare_entry ("Constant shear modulus", "3.3e9",
+                             Patterns::Double (0),
+                             "Shear modulus of material. "
+                             "In case for viscous deformation only model, this parameter is made "
+                             "Reference value is for Europa's icy shell. "
+                             "Units: Pa.");
+          prm.declare_entry ("Constant viscosity", "1e14",
+                             Patterns::Double (0),
+                             "Viscosity of material. "
+                             "Reference value is for viscosity of convecting ice. "
+                             "Units: Pa*s.");
+          prm.declare_entry ("Constant tidal strain rate", "2e-10",
+                             Patterns::Double (0),
+                             "Tidal strain rate on moon. "
+                             "There are ways to implement tidal heating, with strain rate or strain. "
+                             "Currently, this uses strain rate. "
+                             "Reference value is for Europa. "
+                             "Units: 1/s.");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+
+    template <int dim>
+    void
+    ConstantTidalHeating<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Heating model");
+      {
+        prm.enter_subsection("Constant tidal heating");
+        {
+          constant_tidal_frequency = prm.get_double ("Constant tidal frequency");
+          constant_shear_modulus = prm.get_double ("Constant shear modulus");
+          constant_viscosity = prm.get_double ("Constant viscosity");
+          constant_tidal_strain_rate = prm.get_double ("Constant tidal strain rate");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace HeatingModel
+  {
+    ASPECT_REGISTER_HEATING_MODEL(ConstantTidalHeating,
+                                  "constant tidal heating",
+                                  "Implementation of a model in which the tidal heating "
+                                  "rate is constant.")
```

Comment:
```suggestion
                                  "Implementation of a model in which the tidal heating "
                                  "rate is spatially and temporally constant, but depends on a "
                                  "number of parameters that can be set in the input file.")
```
I think it might be useful to state that the default values are chosen for Europa. It would also be useful to link to the publications you mention in your PR.